spec/models/service_spec.rb

```diff
@@ -675,6 +675,31 @@
     end
   end
 
+  describe "#retireable?" do
+    context "with no parent" do
+      context "with type" do
+        service = FactoryBot.build(:service, :type => "thing")
+        it "true" do
+          expect(service.retireable?).to be(true)
+        end
+      end
+
+      context "without type" do
+        service = FactoryBot.build(:service)
+        it "checks type presence" do
+          expect(service.retireable?).to be(false)
+        end
+      end
+    end
+
+    context "with parent" do
+      service = FactoryBot.build(:service, :service => FactoryBot.create(:service))
```

Comment:
These `service` variables shouldn't be created outside of the example.  They can be set up with `let` in the context or moved into the expectation itself since they aren't reused in other examples.